tutorials/intro/plot_object_annotations.py

```diff
@@ -1,245 +1,238 @@
+# the recording began:
+
+print(len(eeglab_raw.annotations))
+print(set(eeglab_raw.annotations.duration))
+print(set(eeglab_raw.annotations.description))
+print(eeglab_raw.annotations.onset[0])
 
 ###############################################################################
-# Save
-# ~~~~
+# More information on working with :class:`~mne.Annotations` objects, including
+# how to add annotations to :class:`~mne.io.Raw` objects interactively, and how
+# to plot, concatenate, load, save, and export :class:`~mne.Annotations`
+# objects can be found in the tutorial :doc:`../raw/plot_annotating_raw`.
 #
-# Note that you can also save annotations to disk in FIF format::
 #
-#     >>> annot.save('my-annot.fif')
+# Converting between Events arrays and Annotations objects
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #
-# Or as CSV with onsets in (absolute) ISO timestamps::
+# Once your experimental events are read into MNE-Python (as either an Events
+# array or an :class:`~mne.Annotations` object), you can easily convert between
+# the two formats as needed. You might do this because, e.g., an Events array
+# is needed for epoching continuous data, or because you want to take advantage
+# of the "annotation-aware" capability of some functions, that automatically
+# omit spans of data if they overlap with certain annotations.
 #
-#     >>> annot.save('my-annot.csv')
-#
-# Or in plain text with onsets relative to ``orig_time``::
-#
-#     >>> annot.save('my-annot.txt')
+# To convert an :class:`~mne.Annotations` object to an Events array, use the
+# function :func:`mne.events_from_annotations` on the :class:`~mne.io.Raw` file
+# containing the annotations. This function will assign an integer Event ID to
+# each unique element of ``raw.annotations.description``, and will return the
+# mapping of descriptions to integer Event IDs along with the derived Event
+# array. By default, one event will be created at the onset of each annotation;
+# this can be modified via the ``chunk_duration`` parameter of
+# :func:`~mne.events_from_annotations` to create equally spaced events within
+# each annotation span. See the function documentation for further details.
+
+events_from_annot, event_dict = mne.events_from_annotations(eeglab_raw)
+print(event_dict)
+print(events_from_annot[:5])
+
+###############################################################################
+# If you want to control which integers are mapped to each unique description
+# value, you can pass a :class:`dict` specifying the mapping as the
+# ``event_id`` parameter of :func:`~mne.events_from_annotations`; this
+# :class:`dict` will be returned unmodified as the ``event_dict``.
 #
+# .. TODO add this when the other tutorial is nailed down:
+#    Note that this ``event_dict`` can be used when creating
+#    :class:`~mne.Epochs` from :class:`~mne.io.Raw` objects, as demonstrated
+#    in :doc:`epoching_tutorial_whatever_its_name_is`.
```

Comment:
In this todo, at some point, we will need to create an example on how to create your own event parser. Maybe this little paragraph should earn more entity or we could just put the entire thing into an example and point to it when we talk about **The What** in [the events and annotations data structures](https://13302-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/intro/plot_object_annotations.html#the-events-and-annotations-data-structures)

Reply:
@massich I don't understand this comment.